### PR TITLE
Cds 269 c holding refinements

### PR DESCRIPTION
**Pull request by mildn01m**

### CODE REVIEW

File: analysis_engine/flight_phase.py [hunk 1]
```diff
@@ -216,25 +217,32 @@ def derive(self, alt_aal=P('Altitude AAL For Flight Phases'),
...
-        turn_rate = rate_of_change(head, HOLDING_MIN_TIME)
+        turn_rate = rate_of_change(head, 150)
         # We scan the entire descent from highest altitude onwards.
         max_alt_to_end = [slices_int(alt_max[0].index, len(alt_aal.array))]
-        above_approach = slices_above(alt_aal.array, 3000)[1]
+        above_approach = slices_above(alt_aal.array, 2000)[1]
         to_scan = slices_and(max_alt_to_end, above_approach)
+        frequency = alt_aal.frequency
+        # Extend each end of band by 5 seconds
+        extend_ends = lambda bands: [
```
Comment (kirku1g) [inline analysis_engine/flight_phase.py]: Use `library.slices_extend_duration()` for this and please add a check for the start to not be negative to the `slices_extend()` function which it calls.

File: analysis_engine/flight_phase.py [hunk 2]
```diff
@@ -216,25 +217,32 @@ def derive(self, alt_aal=P('Altitude AAL For Flight Phases'),
...
+            slice(max(int(x.start - 5 * frequency), 0), min(int(x.stop + 5 * frequency), len(turn_rate))) for x in bands
+        ]
 
         for this_scan in to_scan:
             turn_right_bands = np.ma.clump_unmasked(
                 np.ma.masked_less(turn_rate[slices_int(this_scan)], HOLDING_ROT))
+            turn_right_bands = extend_ends(turn_right_bands)
```
Comment (kirku1g) [inline analysis_engine/flight_phase.py]: See above

File: analysis_engine/flight_phase.py [hunk 3]
```diff
@@ -216,25 +217,32 @@ def derive(self, alt_aal=P('Altitude AAL For Flight Phases'),
...
             turn_bands = shift_slices(slices_remove_small_gaps(turn_right_bands,
                                                                time_limit=HOLDING_MIN_TIME,
-                                                               hz=alt_aal.frequency),
+                                                               hz=frequency),
                                       this_scan.start)
             all_turn_bands.extend(turn_bands)
             turn_left_bands = np.ma.clump_unmasked(
                 np.ma.masked_greater(turn_rate[slices_int(this_scan)], -HOLDING_ROT))
+            turn_left_bands = extend_ends(turn_left_bands)
```
Comment (kirku1g) [inline analysis_engine/flight_phase.py]: See above.

**Comment by crowt01j** [comment]
Requested changes made.

**Pull request merged**